src/widgets/scrollbar.rs

```diff
@@ -446,198 +449,177 @@ impl<'a> Scrollbar<'a> {
-                    area.width,
-                    area.height.saturating_sub(1),
-                )
-            } else {
-                // For horizontal scrollbar, reduce the width by one
-                Rect::new(
-                    area.x + 1,
-                    area.y,
-                    area.width.saturating_sub(1),
-                    area.height,
-                )
-            }
-        } else {
-            area
-        };
-        // Further decrease scrollbar area if an end arrow is present
-        if self.end_symbol.is_some() {
-            if self.is_vertical() {
-                // For vertical scrollbar, reduce the height by one
-                Rect::new(area.x, area.y, area.width, area.height.saturating_sub(1))
-            } else {
-                // For horizontal scrollbar, reduce the width by one
-                Rect::new(area.x, area.y, area.width.saturating_sub(1), area.height)
+    fn get_viewport_len(&self, area: Rect) -> u16 {
+        match self.orientation {
+            ScrollbarOrientation::VerticalRight | ScrollbarOrientation::VerticalLeft => area.height,
+            ScrollbarOrientation::HorizontalBottom | ScrollbarOrientation::HorizontalTop => {
+                area.width
             }
-        } else {
-            area
-        }
-    }
-
-    fn should_not_render(&self, track_start: u16, track_end: u16, content_length: usize) -> bool {
-        if track_end - track_start == 0 || content_length == 0 {
-            return true;
         }
-        false
     }
 
-    fn get_track_start_end(&self, area: Rect) -> (u16, u16, u16) {
+    /// Returns information about scrollbar track
+    ///
+    /// For ScrollbarOrientation::VerticalRight
+    ///
+    /// ```plain
+    ///                   ┌───────── track_axis
+    ///                   v
+    ///   ┌───────────────┐
+    ///   │               ║
+    ///   │               █
+    ///   │               █
+    ///   │               ║
+    ///   └───────────────┘
+    /// ```
+    ///
+    /// For ScrollbarOrientation::HorizontalBottom
+    ///
+    /// ```plain
+    ///   ┌───────────────┐
+    ///   │               │
+    ///   │               │
+    ///   │               │
+    ///   └═══███═════════┘<──────── track_axis
+    /// ```
+    fn get_track_axis(&self, area: Rect) -> u16 {
         match self.orientation {
-            ScrollbarOrientation::VerticalRight => {
-                (area.top(), area.bottom(), area.right().saturating_sub(1))
-            }
-            ScrollbarOrientation::VerticalLeft => (area.top(), area.bottom(), area.left()),
-            ScrollbarOrientation::HorizontalBottom => {
-                (area.left(), area.right(), area.bottom().saturating_sub(1))
-            }
-            ScrollbarOrientation::HorizontalTop => (area.left(), area.right(), area.top()),
+            ScrollbarOrientation::VerticalRight => area.x,
+            ScrollbarOrientation::VerticalLeft => area.x + area.width.saturating_sub(1),
```

Comment:
Seems flipped. Should Left be Right here?

Reply:
Oops my bad. Fixed in recent commit with tests to check for this.